src/CoreWCF.Primitives/src/CoreWCF/Security/IssuedTokenServiceCredential.cs

```diff
@@ -0,0 +1,211 @@
+using System.Collections.Generic;
+using CoreWCF.IdentityModel.Selectors;
+using CoreWCF.IdentityModel.Tokens;
+using CoreWCF;
+using System.Security.Cryptography.X509Certificates;
+using System;
+
+namespace CoreWCF.Security
+{    public class IssuedTokenServiceCredential
+    {
+        internal const bool DefaultAllowUntrustedRsaIssuers = false;
+       // internal const AudienceUriMode DefaultAudienceUriMode = AudienceUriMode.Always;
+        internal const X509CertificateValidationMode DefaultCertificateValidationMode = X509CertificateValidationMode.ChainTrust;
```

Comment:
AudienceUriMode is important. It says whether you care about who the token was issued for. For example if your service is at [https://example.org/MyService.svc](https://example.org/MyService.svc) and someone gives you a token that's been issued for [https://example.com/SomeOtherService.svc](https://example.com/SomeOtherService.svc), then the default is to reject the token but you can make it more relaxed based on the mode.

Reply:
ported..